lib/rbac/filterer.rb

```diff
@@ -717,7 +717,10 @@ def matches_via_descendants(klass, descendant_klass, options)
           klass_id, descendant_id = method_name
           klass.where(klass_id => descendants.select(descendant_id)).distinct
         else
-          MiqPreloader.preload(descendants, method_name)
+          r_klass = descendants.try(:klass) || descendants.first.class
+          if r_klass.reflections.keys.include?(method_name.to_s)
+            MiqPreloader.preload(descendants, method_name)
+          end
```

Comment:
Does it make sense to link this to the `parent_blue_folders` and vice versa?  Is this only here for that one fake association?

Reply:
In other words, "this stuff is here for one specific reason, here's that one place that needs it..."

Reply:
this is here for 2 out of the 3 implementions. `ResourcePool#resource_pool`, and `Vm#_blue_folder`

I did put that in the commit